### PR TITLE
Rewrite content in and after scripts of type text/html

### PR DESCRIPTION
**Pull request by anjackson**
This adds a new test that reproduces the rewriting failure observed for BBC sites (see https://groups.google.com/d/msg/openwayback-dev/E1rMsRBbCf0/EBne7VnYCAAJ).

The test currently fails. The problem appears to stem from the use of a `<script>` of type `text/html`, which seems to confuse the rewriter.

### CODE REVIEW

**Comment by anjackson** [comment]
So, I added a conditional that only marks the parse context as 'being in JavaScript' if the script type is not 'text/html'. This mostly fixes the problem, but it/ adds a close bracket to the CDATA statement, so it's not ideal.

File: wayback-core/src/test/java/org/archive/wayback/archivalurl/FastArchivalUrlReplayParseEventHandlerTest.java [hunk 1]
```diff
@@ -66,6 +66,17 @@ protected void setUp() throws Exception {
...
...</h2> <p class=\"orb-banner-content\" dir=\"ltr\"> The BBC has updated its cookie policy. We use cookies to ensure that we give you the best experience on our website. This includes cookies from third party social media websites if you visit a page which contains embedded content from social media. Such third party cookies may track your use of the BBC website.<span class=\"bbccookies-international-message\"> We and our partners also use cookies to ensure we show you advertising that is relevant to you.</span> If you continue without changing your settings, we'll assume that you are happy to receive all cookies on the BBC website. However, you can change your cookie settings at any time. </p> <ul class=\"orb-banner-options\"> <li id=\"bbccookies-continue\"> <button type=\"button\" id=\"bbccookies-continue-button\">Continue</button> </li> <li id=\"bbccookies-settings\"> <a href=\"http://replay.archive.org/2001/http://www.example.com/privacy/cookies/managing/cookie-settings.html\">Change settings</a> </li> <li id=\"bbccookies-more\"><a href=\"http://replay.archive.org/2001/http://www.example.com/privacy/cookies/bbc\">Find out more</a></li></ul> </div> </div> </section> ]]></script> "
```
Comment (anjackson) [inline wayback-core/src/test/java/org/archive/wayback/archivalurl/FastArchivalUrlReplayParseEventHandlerTest.java]: Note additional ```>``` in ```<![CDATA[ >```.

**Comment by anjackson** [comment]
Hm, the test needs renaming, I guess.

**Comment by ldko** [comment]
I have tested this change, and it worked for me.

**Comment by anjackson** [comment]
Thanks @ldko - much appreciated. I'll try to set up some larger scale testing over the next few days.

**Comment by anjackson** [comment]
Trying to compare what happens here with the IA version is very hard because we've fallen out of sync. Hence #316.

**Pull request merged**